config/default/manager_cloud_env_patch.yaml

```diff
@@ -1,16 +0,0 @@
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: controller-manager
-  namespace: system
-spec:
-  template:
-    spec:
-      containers:
-      - name: manager
-        env:
-          - name: AZURE_ENVIRONMENT
```

Comment:
By removing this we're essentially making it not supported to set the AZURE_ENVIRONMENT value via en env variable right? Wasn't the idea to keep it as optional for back compat?

Reply:
If we keep it we would still need to use envsubst with the manifest, right?

Reply:
Ok but this is a breaking change then for users not using AzurePublicCloud, let's make sure the release note calls that out

Reply:
should we leave this in and just not set a default for it using envsubst? would that be easier for not breaking users

Reply:
> should we leave this in and just not set a default for it using envsubst? would that be easier for not breaking users

that would make it a required variable with clusterctl which might impact even more users (ie. everyone in AzurePublicCloud relying on the default, like quickstart users).

I think removing it and adding a clear release note for non-public cloud users to set the value is the best course of action.